src/state/chainstate.rs

```diff
@@ -216,3 +216,142 @@ impl Drop for ChainstateManagerOptions {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{ChainType, ContextBuilder};
+    use tempdir::TempDir;
+
+    fn create_test_context() -> Context {
+        ContextBuilder::new()
+            .chain_type(ChainType::Regtest)
+            .build()
+            .unwrap()
+    }
+
+    fn create_test_dirs() -> (TempDir, String, String) {
+        let temp_dir = TempDir::new("test_chainman").unwrap();
+        let data_dir = temp_dir.path().to_str().unwrap().to_string();
+        let blocks_dir = format!("{}/blocks", data_dir);
+        (temp_dir, data_dir, blocks_dir)
+    }
+
+    fn create_test_options(
+        context: &Context,
+        data_dir: &str,
+        blocks_dir: &str,
+    ) -> ChainstateManagerOptions {
+        ChainstateManagerOptions::new(context, data_dir, blocks_dir).unwrap()
+    }
+
+    #[test]
+    fn test_chainstate_manager_options_new() {
+        let context = create_test_context();
+        let (_temp_dir, data_dir, blocks_dir) = create_test_dirs();
+
+        let opts = ChainstateManagerOptions::new(&context, &data_dir, &blocks_dir);
+        assert!(opts.is_ok());
+    }
+
+    #[test]
+    fn test_chainstate_manager_options_set_worker_threads() {
```

Comment:
I'm not sure if all these setter tests are useful, since they don't actually exercise anything in a hot path. This just ensures that the program doesn't crash, which in my view should just be a fuzz test. I would keep the basic new tests and those where we actually check errors. The rest seems a bit gratuitous.